test/cypress/integration/card/card-details-page-validation.spec.js

```diff
@@ -0,0 +1,49 @@
+const cardPaymentStubs = require('../../utils/card-payment-stubs')
+
+const tokenId = 'be88a908-3b99-4254-9807-c855d53f6b2b'
+const chargeId = 'ub8de8r5mh4pb49rgm1ismaqfv'
+const createPaymentChargeStubs = cardPaymentStubs.buildCreatePaymentChargeStubs(tokenId, chargeId, 'en')
+const checkCardDetailsStubs = cardPaymentStubs.checkCardDetailsStubs(chargeId)
+
+describe('Card details page validation', () => {
+  beforeEach(() => {
+    // this test is for the full process, the session should be maintained
+    // as it would for an actual payment flow
+    Cypress.Cookies.preserveOnce('frontend_state')
+  })
+
+  it('Should setup the payment and load the page', () => {
```

Comment:
Is this required and intentional ? There is a test already here https://github.com/alphagov/pay-frontend/blob/master/test/cypress/integration/card/payment.spec.js#L77

Reply:
we need this setup for this test to get to the card details page so we can test the errors

Reply:
Can these steps be not part of `Submitting confirmation should show errors` test itself and add the test to `payment.spec.js` ? Not sure of the convention followed for cypress tests

```
cy.task('setupStubs', createPaymentChargeStubs)
cy.visit(`/secure/${tokenId}`)
cy.server()
cy.route('POST', `/check_card/${chargeId}`)
```

Reply:
`payment.spec.js` tests for card number validation. I didn't want to put new validation tests there as there spec is already rather large. I think a better option would be to move the card number validation test from `payment.spec.js` to this new spec.